src/lib/ngsi/ContextAttribute.cpp

```diff
@@ -85,7 +85,7 @@ ContextAttribute::ContextAttribute(ContextAttribute* caP)
   numberValue           = caP->numberValue;
   boolValue             = caP->boolValue;
   compoundValueP        = caP->compoundValueP;
-  caP->compoundValueP   = NULL;
+  caP->compoundValueP   = NULL;   // FIXME P10: really?? shouldn't it be a clone?
```

Comment:
This is a "special" construction, letting the new attribute 'inherit' the compound of the 'older' attribute:

    compoundValueP        = caP->compoundValueP;   // Give compond to new attr
    caP->compoundValueP   = NULL;                  // caP no longer has the compound

This has been tested to work (doesn't mean it will keep working for all future - it clearly depends on its usage).
The advantage with this implementation is that it is **very** fast, the pointer -assignment is immediate. To clone the entire compound and, later on, free not one, but *two* compounds might be the **right** way to do this, but I opted for this 'inheriting-mechanism' instead.

Uptil now it has been enough. If this stops to be true, then we'd have to implement the cloning of the compound and the 'extra free', to avoid leaks.